Engine/Helper.cs

```diff
@@ -656,22 +656,46 @@ public IScriptExtent GetScriptExtentForFunctionName(FunctionDefinitionAst functi
             if (null == functionDefinitionAst)
             {
                 return null;
+            }            
+            var funcNameToken = Tokens.Where(
+                token => ContainsExtent(functionDefinitionAst.Extent, token.Extent) 
+                && token.Text.Equals(functionDefinitionAst.Name));
+            if (funcNameToken.Any())
```

Comment:
Yeah, you'd need to check for null.  The point is that `FirstOrDefault` saves you from two enumerations (which invoke a few methods internally).  `Any()` causes one enumeration to see if there is a first item and `First()` causes another enumeration to get the first item.  Like I said before, it's a very minor concern, I was just relaying a tip I've learned about IEnumerable over time.